File: android/app/src/main/java/com/phicdy/totoanticipation/model/JLeagueService.java
package com.phicdy.totoanticipation.model;


import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.http.GET;

public interface JLeagueService {

    @GET("standings/j1/")
    Call<ResponseBody> j1ranking();

    @GET("standings/j2/")
    Call<ResponseBody> j2ranking();

    @GET("standings/j3/")
    Call<ResponseBody> j3ranking();

    class Factory {
        public static JLeagueService create() {
            Retrofit retrofit = new Retrofit.Builder()
                    .baseUrl("https://www.jleague.jp/")
                    .build();
            return retrofit.create(JLeagueService.class);
        }
    }
}

File: android/legacy/src/main/java/com/phicdy/totoanticipation/legacy/model/JLeagueService.kt
package com.phicdy.totoanticipation.legacy.model


import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.http.GET

interface JLeagueService {

    @GET("standings/j1/")
    fun j1ranking(): Call<ResponseBody>

    @GET("standings/j2/")
    fun j2ranking(): Call<ResponseBody>

    @GET("standings/j3/")
    fun j3ranking(): Call<ResponseBody>

    object Factory {
        fun create(): JLeagueService {
            val retrofit = Retrofit.Builder()
                    .baseUrl("https://www.jleague.jp/")
                    .build()
            return retrofit.create(JLeagueService::class.java)
        }
    }
}
